stumpy/core.py

```diff
@@ -1604,18 +1695,31 @@ def mass_distance_matrix(Q, T, m, distance_matrix, M_T=None, Σ_T=None):
     Σ_T : numpy.ndarray, default None
         Sliding standard deviation of `T`
 
+    T_subseq_isconstant : numpy.ndarray, default None
+        A boolean array that indicates whether a subsequence in `T` is constant (True)
+
     Returns
     -------
         None
     """
-    Q, μ_Q, σ_Q = preprocess(Q, m)
-
-    if M_T is None or Σ_T is None:
-        T, M_T, Σ_T = preprocess(T, m)
+    Q, μ_Q, σ_Q, Q_subseq_isconstant = preprocess(Q, m)
+    if M_T is None or Σ_T is None or T_subseq_isconstant is None:
+        T, M_T, Σ_T, T_subseq_isconstant = preprocess(T, m)
```

Comment:
I don't like this. How would the user override `T_subseq_isconstant` by providing their own? You need to remember that that is the ultimate goal (i.e., allowing the user to specify their own `T_subseq_isconstnat`. In this case, if either  `M_T` or `Σ_T` are `None` then the user provided `T_subseq_isconstant` will be ignored and overridden

Reply:
Even in `mass` we must honor what the user provides

Reply:
@seanlaw 


> How would the user override `T_subseq_isconstant` by providing their own?

This is exactly what came to my mind first. But, then, I tried to change my perspective. I think users should not think of these arguments (i.e. `M_T`, `Σ_T`, `T_subseq_isconstant`) as a way to manipulate them. I think they are just there to speed up things if needed (In fact, that is what I thought before about the purpose of having `M_T` and `Σ_T` arguments  in the `mass` API)

If manipulating `T_subseq_isconstant` is the goal, then I think it would be reasonable to add this argument to `stump` API and many other.

Reply:
@seanlaw 
So, I have been thinking about it more, and I think I can get it now. As you said, we should not change the user input. And, we should just use it if provided. I will revise the code.

Reply:
> we should not change the user input. And, we should just use it if provided.

Exactly! When possible, there should be no surprises.